tokio/tests/rt_common.rs

```diff
@@ -802,7 +820,15 @@ rt_test! {
                         // Yield until connected
                         let mut cnt = 0;
                         while !flag_clone.load(SeqCst){
-                            tokio::task::yield_now().await;
+                            if use_coop {
+                                // Consume a good chunk of budget, which should
+                                // force at least one yield.
+                                for _ in 0..200 {
```

Comment:
We have a fair number of tests that declare a `const BUDGET: usize = 128;` constant. It would be good to do the same here so we can grep for tests that assume what the budget is.